### ISSUE TITLE
Delay Recording process

### ISSUE THREAD

**Issue by agustinmono** [bigbluebutton/bigbluebutton#8945]
I have a problem, for example, I have 80 atendees in a room, the moderator start recording, 1 hour later, the class is over, the moderator ends the session and 10 minutes later it start another one for a different class. So I have half of the server processing the recording, and the other half trying to manage 80 atendees in a room, my question is, is there a way that the recording start processing in non-working hours? like from 10pm to 5am or something like that? 

If not, can I assign a maximum of threads/cores to process the record, so Ill have much more server used in the session.

Thanks a lot

**Comment by mvasylenko** [comment]
> my question is, is there a way that the recording start processing in non-working hours? like from 10pm to 5am or something like that?

You can try to start/stop bbb-record-core.timer via cron.

> If not, can I assign a maximum of threads/cores to process the record, so Ill have much more server used in the session.

You can adjust ffmpeg -threads parameter in
/usr/local/bigbluebutton/core/lib/recordandplayback/generators/video.rb

PS: I had also capacity problems on my bbb setup. The solution was to install multiple bbb servers and scalelite load balancer - https://github.com/blindsidenetworks/scalelite

**Comment by agustinmono** [comment]
Yes i want to install the load balancer, but i cant understand how, can you explain the stepe? I have an issue open in their girhub

**Comment by iSamof** [comment]
The load balancer is essential for addressing immediate requirements. However, it can't address the efficiency issue and better use of resources e.g. during quiet periods, as raised by OP. 

I believe a queue is being developed so recording processes do not run in parallel. 

It would be great if a parameter(s) could allow the setting of when the recording processes start e.g. as OP stated, during certain hours. And if this parameter(s) is adjustable via API, rather than just in a server config, a great deal dynamic control can be exercised. 

An external control tool can then manage priorities, e.g. initiate processing when suitable and suspend when user load increases.

**Comment by mvasylenko** [comment]
> Yes i want to install the load balancer, but i cant understand how, can you explain the steps? I have an issue open in their girhub

I use custom docker-compose script with scalelite, greenlight, redis, postgresql and certbot containers on a single Machine (actually VM), rather then official way. But i will try to help you in your topic at scalelite.

**Comment by mvasylenko** [comment]
> I believe a queue is being developed so recording processes do not run in parallel.

Yes, you are right and this is one of the bottlenecks in production environment. 

> It would be great if a parameter(s) could allow the setting of when the recording processes start e.g. as OP stated, during certain hours. And if this parameter(s) is adjustable via API, rather than just in a server config, a great deal dynamic control can be exercised.

This kind of solution might sound well at first glance, but ... we are talking here just about 2-3 low priority threads. Set correct priorities and let the OS do its job in realtime.

**Comment by basisbit** [comment]
Closing this because all the questions were answered. For any further support questions, please ask at https://groups.google.com/forum/#!forum/bigbluebutton-setup
